check50/__main__.py

```diff
@@ -268,6 +268,12 @@ def main():
     parser.add_argument("-v", "--verbose",
                         action="store_true",
                         help=_("display the full tracebacks of any errors (also implies --log)"))
+    parser.add_argument("--no-download-checks",
+                        action="store_true",
+                        help=_("do not download checks, but use previously downloaded checks instead (only works with --local)"))
```

Comment:
I assume this only works with `--local` because you don't wanna make sure that `.cs50.yaml` is up-to-date before pushing? Should this be enforced by the code? i.e., if you pass `--no-download-checks` without passing `--local`, should there be an error? Or should `--no-download-checks` imply `--local`?

Reply:
It only works with `--local` because we'd otherwise have to put it in a commit message to carry it over to kareem.py. And I don't see any reason to support that, because at that point you should configure the checks differently. 

I don't like `--no-download-checks` implying `--local`. Conceptually this is more of a configuration option on top of `--local`, and not something that changes an entire mode of operation. The way I thought about it, even in `--remote` these options still make sense in that you will never locally download checks and/or install dependencies with these options set. It's just that in `--remote` these wouldn't be installed/downloaded in the first place ;)

Reply:
Right, I totally agree using `--no-download-checks` locally should have no impact on remote `check50`.

> I don't like --no-download-checks implying --local. Conceptually this is more of a configuration option on top of --local, and not something that changes an entire mode of operation. The way I thought about it, even in --remote these options still make sense in that you will never locally download checks and/or install dependencies with these options set. It's just that in --remote these wouldn't be installed/downloaded in the first place ;)

There's either a bit of a contradiction here or maybe I'm confused. So it "only" works with `--local` but it also works with `--remote` and it makes sense there?

I don't think it makes sense with `--remote` because if you specify it in this case, the `.cs50.yaml` file that you have locally could be different from that in the HEAD of the branch (the one that's ultimately going to be used by EKS since this option doesn't impact remote configs) which sounds weird. So for example, local `.cs50.yaml` might be excluding files that remote `.cs50.yaml` is requiring, and yet `check50` would still make a push without the files required by remote if that makes sense?

Reply:
Ah I see the confusion! In `--remote` check50 does not actually clone the checks repo. Instead check50 fetches `.cs50.yml` directly from github.com. We do this for exactly the reason you described :). This is in lib50:

https://github.com/cs50/lib50/blob/f2fb3ae7c9ab0c6d40d924dc5d68be38ce2d500a/lib50/_api.py#L360

Reply:
Oh I see. But this still means that `--no-download-checks --no-install-dependencies` will do nothing if you run remote `check50`, right?

Reply:
Yup, setting or unsetting those flags in `--remote` does nothing because what it touches isn't done in that mode anyway.

Reply:
Perhaps we could change the phrasing from `only works in --local` to `only affects --local`?

Reply:
So they should either imply `--local` or error without it or else they could be a source of confusion. Doesn't it feel weird to you passing options that do nothing? :sweat_smile:

Reply:
It feels weird, but I do think it's right given the alternatives. 

The `--local` implication option is going to cause even more confusion. It feels weirder to me that adding `--no-install-dependencies` changes the (default) mode of operation. Such that this:

````
check50 cs50/problems/2018/x/hello
````

Runs on a remote server, but this:

```
check50 cs50/problems/2018/x/hello --no-install-dependencies
```

Now runs on your machine because it's not installing dependencies?

Alternatively, we could error in the latter case. I'm not strongly opposed to this, because it is a useless flag after all. But, technically, what the flag tells you is still true. It is not installing dependencies (on your local machine). 

Perhaps then we should issue a warning? `"setting --no-install-dependencies has no effect without also setting --local"`

Reply:
Fair. At this point, if we decide that `--no-*` imply `--local`, I would show a warning or something that says that `check50` will be running in local mode if `--local` is not explicitly specified or perhaps the easier approach would be just exit with an error saying that this is only allowed in local mode instead of just silently ignoring the flags. Maybe I'm over-complicating things though. Any thoughts @crossroads1112 @dmalan @brianyu28?

Reply:
Fyi, from Docker's docs: `--memory-swap is a modifier flag that only has meaning if --memory is also set`

https://docs.docker.com/config/containers/resource_constraints/#--memory-swap-details

Reply:
Right, but that's enforced?

```
$ docker run -it --rm --memory-swap 100m alpine
docker: Error response from daemon: You should always set the Memory limit when using Memoryswap limit, see usage.
See 'docker run --help'.
```

Reply:
+1 for no-install-dependencies not implying local but displaying a warning if passed without it.

Reply:
Went with the following warning:

```
Warning: --no-downloads-checks and --no-install-dependencies have no effect without --local
```